server.js

```diff
@@ -23,6 +41,37 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.json(thoughts)
+})
+
+app.post('/thoughts', async (req, res) => {
+  // retrive info sent by client
+  const { message } = req.body;
+  // use mongoose model to create the database entry
+  const thought = new Thought({ message })
+
+  try {
+    //success
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {
+    res.status(400).json({ message: 'could not save thought to the database', error: err.errors })
+  }
+})
+
+app.post('/thoughts/:id/like', async (req, res) => {
+  try {
+    const savedLike = await Thought.updateOne(
+      { _id: req.params.id }, { $inc: { hearts: 1 } }
+    )
+    res.json(savedLike)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not like post, id not found', error: err.errors })
+  }
+})
+
```

Comment:
This one should be `/:thoughtId/like`

Reply:
to make it more RESTful?

Reply:
To meet the requirements. https://www.notion.so/Project-Happy-Thoughts-API-3dc1710667304987894f954dad265fc9